### PR TITLE
Support Access Credentials

### PR DESCRIPTION
**Pull request by futuretea**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Related Issue:**
https://github.com/harvester/harvester/issues/1538

**Test plan:**

### CODE REVIEW

File: pkg/api/vm/handler.go [hunk 1]
```diff
@@ -449,10 +449,24 @@ func (h *vmActionHandler) createTemplate(namespace, name string, input CreateTem
...
-	return h.createCloudConfigSecrets(vmtv, vm)
+	return h.createSecrets(vmtv, vm)
 }
 
-func (h *vmActionHandler) createCloudConfigSecrets(templateVersion *harvesterv1.VirtualMachineTemplateVersion, vm *kv1.VirtualMachine) error {
+func (h *vmActionHandler) createSecrets(templateVersion *harvesterv1.VirtualMachineTemplateVersion, vm *kv1.VirtualMachine) error {
+	for _, credential := range vm.Spec.Template.Spec.AccessCredentials {
+		if sshPublicKey := credential.SSHPublicKey; sshPublicKey != nil && sshPublicKey.Source.Secret != nil {
+			toCreateSecretName := getTemplateVersionSSHPublicKeySecretName(templateVersion.Name)
+			if err := h.copySecret(sshPublicKey.Source.Secret.SecretName, toCreateSecretName, templateVersion); err != nil {
```
Comment (gitlawr) [inline pkg/api/vm/handler.go]: There could be multiple secrets in AccessCredentials. It seems like we create the copies using the same secret name.

Comment (futuretea) [reply pkg/api/vm/handler.go]: fixed in https://github.com/harvester/harvester/pull/1789/commits/0b2109523c852de4be515fa74019184d168f08e1

File: pkg/controller/master/backup/util.go [hunk 2]
```diff
@@ -14,6 +16,12 @@ import (
 	"github.com/harvester/harvester/pkg/settings"
 )
 
+const (
+	// UI stores the mapping between access credential secret and ssh keys in the annotation `harvesterhci.io/dynamic-sshkeys-names`
+	// example: '{"secretname1":["sshkeyname1","sshkeyname2"],"secretname2":["sshkeyname3","sshkeyname4"]}'
+	dynamicSSHKeysNamesAnnotation = "harvesterhci.io/dynamic-sshkeys-names"
```
Comment (gitlawr) [inline pkg/controller/master/backup/util.go]: Is `sshkeyname1` the name of a keypair resource? why secretname maps to multiple sshkeynames

Comment (futuretea) [reply pkg/controller/master/backup/util.go]: Yes, `sshkeyname1` is a example name of a keypair resource.
One secret can contain multiple public keys. 
reference: https://kubevirt.io/user-guide/virtual_machines/accessing_virtual_machines/#dynamic-ssh-key-injection-via-qemu-user-agent

**Review by gitlawr — APPROVED** [review]
LGTM

File: pkg/controller/master/backup/util.go (hunk 2 above)
Comment (guangbochen) [inline pkg/controller/master/backup/util.go]: prefer to name it `dynamic-ssh-key-names`

Comment (futuretea) [reply pkg/controller/master/backup/util.go]: CC @DaiYuzeng

Comment (futuretea) [reply pkg/controller/master/backup/util.go]: Updated

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**